ext/standard/math.c

```diff
@@ -668,7 +668,8 @@ PHP_FUNCTION(log)
 	}
 
 	if (base == 1.0) {
-		RETURN_DOUBLE(ZEND_NAN);
+        zend_argument_value_error(2, "must not be 1");
+        RETURN_THROWS();
 	}
 
 	if (base <= 0.0) {
```

Comment:
Might make sense to unify both conditions and messages?

Reply:
Given that values in `]0;1[` work, I'm not sure. I like having an accurate error message, however I'm not against having "must not be 1 or less than 0" if you prefer.